src/test/foldingProvider.test.ts

```diff
@@ -140,6 +140,92 @@ endif::[]`)
   ])
 })
 
+//test folding on block code
+
+test('Should fold open block', () => {
+  const folds = getFoldsForDocument(
+    `this is a paragraph
+
+--
+An open block can be an anonymous container,
+or it can masquerade as any other block.
+--
+
+this is a paragraph`)
+  assert.strictEqual(folds.length, 1, 'expecting 1 fold')
+  assert.deepStrictEqual(folds, [
+    new vscode.FoldingRange(2, 5, vscode.FoldingRangeKind.Region),
+  ])
+})
+
+test('Should fold open block from dashes to the end of the document if no end block dashes', () => {
+  const folds = getFoldsForDocument(
+    `this is a paragraph
+
+--
+An open block can be an anonymous container,
+or it can masquerade as any other block.
+
+this is a paragraph`)
+  assert.strictEqual(folds.length, 1, 'expecting 1 fold')
+  assert.deepStrictEqual(folds, [new vscode.FoldingRange(2, 6, vscode.FoldingRangeKind.Region)])
+})
+
+test('Should fold open block acting as a sidebar', () => {
+  const folds = getFoldsForDocument(
+    `this is a paragraph
+
+[sidebar]
+.Related information
+--
+This is aside text.
+
+It is used to present information related to the main content.
+--
+
+this is a paragraph`)
+  assert.strictEqual(folds.length, 2, 'expecting 2 folds')
+  assert.deepStrictEqual(folds, [
+    new vscode.FoldingRange(2, 4, vscode.FoldingRangeKind.Region),
+    new vscode.FoldingRange(4, 8, vscode.FoldingRangeKind.Region),
+  ])
+})
+
+test('Should fold open block acting as a source block', () => {
+  const folds = getFoldsForDocument(
+    `this is a paragraph
+
+[source]
+--
+puts "I'm a source block!"
+--
+
+this is a paragraph`)
+  assert.strictEqual(folds.length, 1, 'expecting 1 fold')
+  assert.deepStrictEqual(folds, [
+    new vscode.FoldingRange(2, 5, vscode.FoldingRangeKind.Region),
+  ])
+})
+
+test('Nested open blocks  should behave like 2 separate blocks', () => {
+  const folds = getFoldsForDocument(
+    `this is a paragraph
+
+[source]
+--
+--
+puts "I'm a nested block!"
+--
+--
+
+this is a paragraph`)
+  assert.strictEqual(folds.length, 2, 'expecting 2 folds')
+  assert.deepStrictEqual(folds, [
```

Comment:
I don't think this is a good test as the expected result is unclear.
In this case, Asciidoctor will produce the following blocks:

- An empty listing/source block
- A paragraph that contains "puts "I'm a nested block!"
- An empty open block

Reply:
I take it back, the result is not unclear, the parsing rules are:

- An empty open block masquerading as a source block:
```
[source]
--
--
```

- A paragraph that contains "puts "I'm a nested block!"
```
puts "I'm a nested block!"
```
- An empty open block

```
--
--
```

- A paragraph

```

this is a paragraph
```